### ISSUE TITLE
[proposition] Serialisation of standard types

### ISSUE THREAD

**Issue by frankiezafe** [djiamnot/gdosc#5]
In the idea of [VVVV osc](https://github.com/vvvv/vvvv-sdk/blob/develop/common/src/core/Utils/OSC/OSCMessage.cs) implementation, adding methods to pack and unpack vector2, vector3, vector4, quaternion, matrix3x3 and matrix4x4, with or without adding special types in the OSC messages.

For instance, you do:
```python
var va = Vector3(1,2,3)
var vb = Vector3(4,5,6)
setAddress("/test")
appendV3(va)
appendV3(vb)
```
to send and to retrieve, you do: 
```python
var va = msg.getAsV3(0)
var vb = msg.getAsV3(3)
```
If no new types are added, the vectors and matrices are sent as an array of floats. Therefore, it is always possible to retrieve them by doing:
```python
var x = msg.getAsFloat(0)
var y = msg.getAsFloat(1)
var z = msg.getAsFloat(2)
```
I did something like that in another implementation, and it is quite practical to serialise info from the 3d world.

**Comment by djiamnot** [comment]
Yeah, it's definitely worth it. Ultimately, it should support the OSC spec (http://opensoundcontrol.org/spec-1_0) and utility function specific for Godot are definitely useful, including serialization.

**Comment by frankiezafe** [comment]
sending part is done in OSCsender for vector2, vector3, quaternion and transform
